docs: update drag and drop tutorial (#29200)
electron/electron: docs/tutorial/native-file-drag-drop.md

## Example

> A simple example demonstrating how you can create a file on the fly to be dragged out of the window.

Reviewer: ```suggestion
An example demonstrating how you can create a file on the fly to be dragged out of the window.
```

I think we should not use terms like "simple", "easy", etc. in the documentation. People might understand what's going on at different speeds and if they need to read the docs the task at hand might not be easy already.

Jim Fischer gave a talk a few years back about this: [Don't say "simply"](https://www.youtube.com/watch?v=gsT2BBWBVmM).

### Preload.js

> In `preload.js` use the [`contextBridge`] to inject a method `window.electron.startDrag(...)` that will trigger an ipc message from to the main process.

Reviewer: ```suggestion
In `preload.js` use the [`contextBridge`] to inject a method `window.electron.startDrag(...)` that will send an ipc message to the main process.
```

```js
const { contextBridge, ipcRenderer } = require('electron')
const fs = require('fs')

contextBridge.exposeInMainWorld('electron', {
  startDrag: (fileName) => {
    // Create a new file to copy - you can also copy existing files.
    fs.writeFileSync(fileName, '# Test drag and drop')

    ipcRenderer.send('ondragstart', process.cwd() + `/${fileName}`)
  }
})
```

### Index.html

Add a draggable element to `index.html`, and reference your renderer script:

```html
<div style="border:2px solid black;border-radius:3px;padding:5px;display:inline-block" draggable="true" id="drag">Drag me</div>
<script src="renderer.js"></script>
```

### Renderer.js

In `renderer.js` setup the renderer process to handle drag events by calling the method you added via the [`contextBridge`] above.

```javascript
document.getElementById('drag').ondragstart = (event) => {
  event.preventDefault()
  window.electron.startDrag('drag-and-drop.md')
}
```

### Main.js

> In the Main process(`main.js` file), expand the received event with a path to the file that is

Reviewer: Let's fix this while we are doing changes here 😊
```suggestion
In the Main process (`main.js` file), expand the received event with a path to the file that is
```

## Example

An example demonstrating how you can create a file on the fly to be dragged out of the window.

### Preload.js

> In `preload.js` use the [`contextBridge`] to inject a method `window.electron.startDrag(...)` that will send an ipc message to the main process.

Reviewer 2: ```suggestion
In `preload.js` use the [`contextBridge`] to inject a method `window.electron.startDrag(...)` that will send an IPC message to the main process.
```

```js
const { contextBridge, ipcRenderer } = require('electron')
const fs = require('fs').promises
const path = require('path')

contextBridge.exposeInMainWorld('electron', {
  startDrag: async (fileName) => {
    // Create a new file to copy - you can also copy existing files.
    await fs.writeFile(fileName, '# Test drag and drop')

    ipcRenderer.send('ondragstart', path.join(process.cwd(), fileName))
  }
})
```

### Index.html

Add a draggable element to `index.html`, and reference your renderer script:

```html
<div style="border:2px solid black;border-radius:3px;padding:5px;display:inline-block" draggable="true" id="drag">Drag me</div>
<script src="renderer.js"></script>
```

### Renderer.js

> In `renderer.js` setup the renderer process to handle drag events by calling the method you added via the [`contextBridge`] above.

Reviewer 2: ```suggestion
In `renderer.js` set up the renderer process to handle drag events by calling the method you added via the [`contextBridge`] above.
```
